pkg/apis/proxy/register_test.go

```diff
@@ -0,0 +1,86 @@
+package proxy
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+func TestGetBackend(t *testing.T) {
+	backend1 := newTestServer()
+	defer backend1.server.Close()
+	w := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(w)
+	ctx.Request = httptest.NewRequest(http.MethodGet, "/backends", bytes.NewBufferString("hello"))
+	proxy, err := NewSprayProxy(false, true, zap.NewNop(), backend1.server.URL)
+	if err != nil {
+		t.Fatalf("failed to set up proxy: %v", err)
+	}
+	proxy.GetBackends(ctx)
+	if w.Code != http.StatusOK {
+		t.Errorf("expected status code %d, got %d", http.StatusOK, w.Code)
+	}
+	expected := backend1.server.URL
+	responseBody := w.Body.String()
+	if !strings.Contains(responseBody, expected) {
+		t.Errorf("expected string %q did not appear in %q", expected, responseBody)
+	}
+
+	if backend1.err != nil {
+		t.Errorf("backend 1 error: %v", backend1.err)
+	}
+}
+
+func TestResiterBackendLog(t *testing.T) {
```

Comment:
nit - spelling (not blocking)

```suggestion
func TestRegsiterBackendLog(t *testing.T) {
```